compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -87,6 +87,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             } else if lang_items.sized_trait() == Some(def_id) {
                 // Sized is never implementable by end-users, it is
                 // always automatically computed.
+
+                // Error type cannot possibly implement `Sized` (fixes #123154)
+                if let Err(e) = obligation.predicate.skip_binder().self_ty().error_reported() {
+                    return Err(SelectionError::Overflow(e.into()));
+                }
```

Comment:
add a `FIXME` so we'll try moving this all the way up to right after the `obligation.predicate.skip_binder().self_ty().is_ty_var()` check. This may be useful for all trait obligations.

Reply:
Done